src/components/useIntl.js

```diff
@@ -0,0 +1,9 @@
+import {useContext} from 'react';
+import {IntlContext} from './../react-intl';
+import {invariantIntlContext} from '../utils';
+
+export default function useIntl() {
+  const intl = useContext(IntlContext);
+  invariantIntlContext({ intl });
+  return intl;
```

Comment:
As I suggested in the issue related to this PR, what do you guys think about changing this to
```js
return [intl.formatMessage, intl];

// Allows for usage as following
const [ t ] = useIntl()

// Currently
const { formatMessage: t } = useIntl()
```

I think this might allow for slightly less code, as 90% of the time people use `formatMessage`.
However it might take away some simplicity.  I think it'd be worth it.

Reply:
I personally think it is beautiful, I'm totally okay with this. As for the simplicity, I think this is also cleaner and idiomatic (think about other lib and the primitive hooks' API), but it surly need more documentation.

Are we agree it is the convenience we want?! I think it's straightforward.

Reply:
I agree. I think it's a good way to go.

Reply:
@DragonRaider5, I've updated my PR, please have a look 😸!